File: public/tools/elements/profile/GtProfile.ts
import {TabsGenerator, View} from "../../client/router/View";
import {Element, Property} from "../../polymer/Annotations";
import {ProfileUser} from "ProfileUser";
import {ProfileInfo} from "./ProfileInfo";
import {ProfileChars} from "./ProfileChars";
import {PolymerElement} from "../../polymer/PolymerElement";

const ProfileTabs: TabsGenerator = (view, path, user) => [
	{ title: "Profile", link: "/profile", active: view == GtProfile }
];

@View("profile", ProfileTabs)
@Element({
	selector: "gt-profile",
	template: "/assets/views/profile.html",
	dependencies: [ProfileUser, ProfileInfo, ProfileChars]
})
export class GtProfile extends PolymerElement {
	@Property
	public user: number = this.app.user.id;
}

File: client/src/gt/component/profile/GtProfile.scala
package gt.component.profile

import gt.component.widget.GtBox
import gt.component.widget.form.GtButton
import gt.component.{GtHandler, Tab, View}
import util.jsannotation.js
import xuen.Component

object GtProfile extends Component[GtProfile](
	selector = "gt-profile",
	templateUrl = "/assets/imports/views/profile.html",
	dependencies = Seq(ProfileUser, ProfileInfo, ProfileChars, GtBox, GtButton)
) with View {
	val module = "profile"
	val tabs: TabGenerator = (selector, path, user) => Seq(
		Tab("Profile", "/profile", active = true)
	)
}

@js class GtProfile extends GtHandler {
	val user = attribute[Int] := app.user.id

	/** Whether the profile is editable by the current user */
	val editable = user ~ { id => id == app.user.id || app.user.promoted }
}


